src/create-config.ts

```diff
@@ -335,7 +336,9 @@ const main = async () => {
     },
   };
 
-  writeOperationsRepository(updatedOpsData);
+  writeOperationsRepository(updatedOpsData, operationRepositoryTarget);
 };
 
-runAndHandleErrors(main);
+if (require.main === module) runAndHandleErrors(main);
```

Comment:
It might be a good idea to put a comment explaining this check (I had to google it 😅 ) and also explain why we need this check.

Maybe instead of doing that check you could move all the logic within `main` to a new function that will be exported and call by `main`. Then the tests can call that function instead. That's just a thought...

Reply:
BTW it seems that this check is being done before all calls to `runAndHandleErrors` so should we move the check inside the function?

Reply:
> It might be a good idea to put a comment explaining this check (I had to google it 😅 ) and also explain why we need this check.

haha yeah, this was actually quite the problem to solve. For some reason the tests were calling createConfig twice when I did not do this. The only explanation I can give is that the tests are importing the script and while importing it ends up calling the script once and then again when I invoke it it. Very weird .

> Maybe instead of doing that check you could move all the logic within main to a new function that will be exported and call by main. Then the tests can call that function instead. That's just a thought...

I'll try doing this and see if that problem occurs


> BTW it seems that this check is being done before all calls to runAndHandleErrors so should we move the check inside the function?

Thats a good idea

Reply:
Yeah, interesting, TIL, everything gets evaluated on import because function definitions are code executions:
`const myFunc = () => {};` is an executable command to Node saying "create a function and store it in `myFunc`".

So you generally have to split out your logic from the main entrypoint if you're planning on writing tests or you're planning on re-using those functions elsewhere.

Reply:
> BTW it seems that this check is being done before all calls to runAndHandleErrors so should we move the check inside the function?

So I tested this and it won't work because the check like @aquarat has stated checks if the function was executed as the main module.. If we add the check to run `runAndHandleErrors ` then the check always fails because that was not the script that was invoked.

Reply:
This might be an alternative (?):
```
export const runAndHandleErrors = (fn: () => Promise<unknown>) => {
  if (process.env.JEST_WORKER_ID) {
    return;
  }
...
```

Reply:
I think this is fine :)